### PR TITLE
Setting autocommit to false before carrying out connection.commit

### PR DESCRIPTION
**Pull request by msm1992**
### Proposed changes in this pull request

* Setting autocommit to false before carrying out connection.commit in CertificateMgtDAO.java

### When should this PR be merged

ASAP

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/CertificateMgtDAO.java [hunk 1]
```diff
@@ -114,26 +114,35 @@ public boolean isTableExists() throws CertificateManagementException {
...
+                APIMgtDBUtil.closeStatement(preparedStatement);
+                APIMgtDBUtil.closeAllConnections(preparedStatement, connection, resultSet);
+                connection.setAutoCommit(initialAutoCommit);
             }
         } catch (SQLException e) {
             if (log.isDebugEnabled()) {
-                log.debug("Error occurred while retrieving certificate metadata. ", e);
+                log.debug("Error occurred in finally block while retrieving certificate metadata. ", e);
             }
-            handleException("Error while retrieving certificates.", e);
```
Comment (uthaiyashankar) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/CertificateMgtDAO.java]: We are removing this line? Which means, the exception is swallowed if debug is not enabled?

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/CertificateMgtDAO.java [hunk 2]
```diff
@@ -169,34 +178,42 @@ public boolean addCertificate(String alias, String endpoint, int tenantId) throw
...
+                APIMgtDBUtil.closeStatement(preparedStatement);
+                APIMgtDBUtil.closeAllConnections(preparedStatement, connection, null);
+                connection.setAutoCommit(initialAutoCommit);
             }
+        } catch (SQLException e) {
             if (log.isDebugEnabled()) {
-                log.debug("Error occurred while adding certificate metadata to database.", e);
+                log.debug("Error occurred in finally block while adding certificate metadata. ", e);
             }
-            handleException("Error while persisting certificate metadata.", e);
```
Comment (uthaiyashankar) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/CertificateMgtDAO.java]: Exception swallowed?

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/CertificateMgtDAO.java [hunk 3]
```diff
@@ -220,23 +237,32 @@ private CertificateMetadataDTO getCertificate(String alias, String endpoint) thr
...
+                }
+            } catch (SQLException e) {
+                handleException("Error while retrieving certificate metadata.", e);
+            } finally {
+                APIMgtDBUtil.closeStatement(preparedStatement);
+                APIMgtDBUtil.closeAllConnections(preparedStatement, connection, resultSet);
+                connection.setAutoCommit(initialAutoCommit);
             }
         } catch (SQLException e) {
-            handleException("Error while retrieving certificate metadata.", e);
```
Comment (uthaiyashankar) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/CertificateMgtDAO.java]: Exception swallowed?

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/CertificateMgtDAO.java [hunk 4]
```diff
@@ -259,24 +285,33 @@ public CertificateMetadataDTO getCertificate(String alias, String endpoint, int
...
```
Comment (uthaiyashankar) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/CertificateMgtDAO.java]: Exception swallowed?

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/CertificateMgtDAO.java (hunk 1 above)
Comment (msm1992) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/CertificateMgtDAO.java]: fixed with https://github.com/wso2/carbon-apimgt/commit/3c8d1e61dc642ebe8179c483156798312a03382b

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/CertificateMgtDAO.java (hunk 2 above)
Comment (msm1992) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/CertificateMgtDAO.java]: fixed with https://github.com/wso2/carbon-apimgt/commit/3c8d1e61dc642ebe8179c483156798312a03382b

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/CertificateMgtDAO.java (hunk 3 above)
Comment (msm1992) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/CertificateMgtDAO.java]: fixed with https://github.com/wso2/carbon-apimgt/commit/3c8d1e61dc642ebe8179c483156798312a03382b

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/CertificateMgtDAO.java (hunk 4 above)
Comment (msm1992) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/CertificateMgtDAO.java]: fixed with https://github.com/wso2/carbon-apimgt/commit/3c8d1e61dc642ebe8179c483156798312a03382b

**Pull request merged**